dali/operators/reader/loader/video_loader.h

```diff
@@ -176,19 +186,37 @@ class VideoLoader : public Loader<GPUBackend, SequenceWrapper> {
   void PrepareMetadataImpl() override {
     int total_count = 1 + (count_ - 1) * stride_;
 
-    for (size_t i = 0; i < file_label_pair_.size(); ++i) {
-      const auto& file = get_or_open_file(file_label_pair_[i].first);
+    for (size_t i = 0; i < file_info_.size(); ++i) {
+      const auto& file = get_or_open_file(file_info_[i].video_file);
       const auto stream = file.fmt_ctx_->streams[file.vid_stream_idx_];
       int frame_count = file.frame_count_;
 
-      for (int s = 0; s < frame_count && s + total_count <= frame_count; s += step_) {
-        frame_starts_.emplace_back(sequence_meta{i, s, file_label_pair_[i].second,
+      int start_frame = 0;
+      int end_frame = file.frame_count_;
+      float start = file_info_[i].start_time;
+      float end = file_info_[i].end_time;
+      if (file_list_frame_num_) {
+        start_frame = start;
+        end_frame = end;
+        DALI_ENFORCE(end_frame <= file.frame_count_, "End frame number is greater than "
+            "total number of frames for file " + file_info_[i].video_file);
+      } else if (!(start == -1 && end == -1)) {
```

Comment:
```suggestion
      } else if (start != -1 && end != -1) {
```
I think both start and end need to != -1, not only one of them